lensless/trainable_recon.py

```diff
@@ -0,0 +1,143 @@
+# #############################################################################
+# trainable_recon.py
+# =================
+# Authors :
+# Yohann PERRON [yohann.perron@gmail.com]
+# #############################################################################
+
+import abc
+from lensless.recon import ReconstructionAlgorithm
+
+try:
+    import torch
+
+except ImportError:
+    raise ImportError("Pytorch is require to use trainable reconstruction algorithms.")
+
+
+class TrainableReconstructionAlgorithm(ReconstructionAlgorithm, torch.nn.Module):
+    """
+    Abstract class for defining lensless imaging reconstruction algorithms with trainable parameters.
+
+    The following abstract methods need to be defined:
```

Comment:
As it inherits from `ReconstructionAlgorithm` do you need to include rest of docstring?

Should include mention of new abstraction method `batch_call`